libr/core/cmd_seek.c

```diff
@@ -273,30 +273,41 @@ static void seek_to_register(RCore *core, const char *input, bool is_silent) {
 }
 
 static int cmd_seek_opcode_backward(RCore *core, int n) {
-	int val = 0;
+	int i, val = 0;
 	// N previous instructions
 	ut64 addr = core->offset;
 	int ret = 0;
 	int numinstr = n * -1;
 	if (r_core_prevop_addr (core, core->offset, numinstr, &addr)) {
 		ret = core->offset - addr;
+		r_core_seek (core, addr, true);
 	} else {
 #if 0
 		// core_asm_bwdis_len is buggy as hell we should kill it. seems like prevop_addr
 		// works as expected, because is the one used from visual
 		ret = r_core_asm_bwdis_len (core, &instr_len, &addr, numinstr);
 #endif
-		ut64 prev_addr = prevop_addr (core, core->offset);
-		if (prev_addr <= core->offset) {
-			RAsmOp op;
+		const int mininstrsize = r_anal_archinfo (core->anal, R_ANAL_ARCHINFO_MIN_OP_SIZE);
+		ut64 addr = core->offset;
+		for (i = 0; i < numinstr; i++) {
+			ut64 prev_addr = prevop_addr (core, addr);
+			if (prev_addr == UT64_MAX) {
+				prev_addr = addr - mininstrsize;
+			}
+			if (prev_addr >= core->offset) {
```

Comment:
can this ever happen? i wouldn remove this (ideally i would put an `assert`, because this should never happen, otherwise it means there's an error in `prevop_addr` and not in this function)

Reply:
academically this is true, the problem is that if someone updates prevop_addr or the current one fails somehow to result in such undesired behaviour the user will get a shell and loose all the job he have been doing during 2h in r2. that's why i reject the use of asserts. also, coverity should be able to identify this as dead code if so.